src/equations/1d/compressible_euler.jl

```diff
@@ -0,0 +1,588 @@
+                                              surface_flux_function,
+                                              equation::CompressibleEulerEquations1D)
+  u_boundary = initial_conditions_convergence_test(x, t, equation)
+
+  # Calculate boundary flux
+  if direction == 2 # u_inner is "left" of boundary, u_boundary is "right" of boundary
+    flux = surface_flux_function(u_inner, u_boundary, orientation, equation)
+  else # u_boundary is "left" of boundary, u_inner is "right" of boundary
+    flux = surface_flux_function(u_boundary, u_inner, orientation, equation)
+  end
+
+  return flux
+end
+
+
+# Apply source terms
+function source_terms_convergence_test(ut, u, x, element_id, t, n_nodes, equation::CompressibleEulerEquations1D)
+  # Same settings as in `initial_conditions`
+  c = 2
+  A = 0.1
+  L = 2
+  f = 1/L
+  ω = 2 * pi * f
+  γ = equation.gamma
+
+  for i in 1:n_nodes
+    x1 = x[1, i, element_id]
+
+    si, co = sincos((t - x1)*ω)
+    tmp = (-((4 * si * A - 4c) + 1) * (γ - 1) * co * A * ω) / 2
+
+    ut[2, i, element_id] += tmp
+    ut[3, i, element_id] += tmp
+
+    # Original terms (without performanc enhancements)
+    # ut[1, i, element_id] += 0
+    # ut[2, i, element_id] += (-(((4 * sin((t - x1) * ω) * A - 4c) + 1)) *
+    #                          (γ - 1) * cos((t - x1) * ω) * A * ω) / 2
+    # ut[3, i, element_id] += (-(((4 * sin((t - x1) * ω) * A - 4c) + 1)) *
+    #                          (γ - 1) * cos((t - x1) * ω) * A * ω) / 2
+  end
+
+  return nothing
+end
+#=
+function source_terms_eoc_test_coupled_euler_gravity(ut, u, x, element_id, t, n_nodes, equation::CompressibleEulerEquations2D)
+  # Same settings as in `initial_conditions_eoc_test_coupled_euler_gravity`
+  c = 2.0
+  A = 0.1
+  G = 1.0 # gravitational constant, must match coupling solver
+  C_grav = -2.0 * G / pi # 2 == 4 / ndims
+
+  for j in 1:n_nodes, i in 1:n_nodes
+    x1 = x[1, i, j, element_id]
+    x2 = x[2, i, j, element_id]
+    si, co = sincos(pi * (x1 + x2 - t))
+    rhox = A * pi * co
+    rho  = c + A *  si
+
+    ut[1, i, j, element_id] += rhox
+    ut[2, i, j, element_id] += rhox
+    ut[3, i, j, element_id] += rhox
+    ut[4, i, j, element_id] += (1.0 - C_grav*rho)*rhox
+  end
+
+  return nothing
+end
+
+function source_terms_eoc_test_euler(ut, u, x, element_id, t, n_nodes, equation::CompressibleEulerEquations2D)
+  # Same settings as in `initial_conditions_eoc_test_coupled_euler_gravity`
+  c = 2.0
+  A = 0.1
+  G = 1.0
+  C_grav = -2 * G / pi # 2 == 4 / ndims
+
+  for j in 1:n_nodes, i in 1:n_nodes
+    x1 = x[1, i, j, element_id]
+    x2 = x[2, i, j, element_id]
+    si, co = sincos(pi * (x1 + x2 - t))
+    rhox = A * pi * co
+    rho  = c + A *  si
+
+    ut[1, i, j, element_id] += rhox
+    ut[2, i, j, element_id] += rhox * (1 -     C_grav * rho)
+    ut[3, i, j, element_id] += rhox * (1 -     C_grav * rho)
+    ut[4, i, j, element_id] += rhox * (1 - 3 * C_grav * rho)
+  end
+
+  return nothing
+end
+=#
```

Comment:
```suggestion
```
Remove commented code - if wee need to recover it, we can just look at the 2D solver.